### PR TITLE
Ensure that we can change the local rotation when anchoring

### LINKED ISSUE #13183: Anchored structures remain tilted
## Description

Dragged a structure out of a salvage grid, it ended up tilted inside the shuttle even after being anchored
**Reproduction**

**Screenshots**

image

**Additional context**

### PR DESCRIPTION
**Pull request by eoineoineoin**
## About the PR 

Always allow local rotation changes when anchoring an object. Fixes #13183. Not 100% sure what the use-case for the NoLocalRotation setting is for -- those objects probably need a more general fix for when they start to overlap a new grid, but this change at least allows players to anchor an object to realign it's transform so that it fits through doors etc.

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- fix: Anchoring large objects now aligns them to their new grid

### CODE REVIEW

File: Content.Server/Construction/AnchorableSystem.cs [hunk 1]
```diff
@@ -83,7 +83,14 @@ private void OnAnchorComplete(EntityUid uid, AnchorableComponent component, TryA
...
             var rot = xform.LocalRotation;
-            xform.LocalRotation = Math.Round(rot / (Math.PI / 2)) * (Math.PI / 2);
+            var snappedAngle = Math.Round(rot / (Math.PI / 2)) * Math.PI * 2;
+            // We want to override the local rotation to `snappedAngle` regardless
+            // of the xforms `NoLocalRotation` setting, so temporarily allow a
+            // local rotation and restore it
+            var oldLocalRotation = xform.NoLocalRotation;
+            xform.NoLocalRotation = false;
+            xform.LocalRotation = snappedAngle;
+            xform.NoLocalRotation = oldLocalRotation;
```
Comment (metalgearsloth) [inline Content.Server/Construction/AnchorableSystem.cs:93]: This defeats the point of having no rotation?

Comment (eoineoineoin) [reply Content.Server/Construction/AnchorableSystem.cs:93]: I agree that it feels kinda clumsy, but this actually _keeps_ the rotation; the issue is that if you have an object with NoLocalRotation and move it to another grid whose basis isn't aligned with the basis of the grid that the object was created in, you can never rotate that object. This at least causes the rotation to get snapped (as the code was already attempting to do) to some alignment on new grid.

Comment (ElectroJr) [reply Content.Server/Construction/AnchorableSystem.cs:93]: The real issue is whatever is giving it a non-zero local rotation in the first place.

Comment (ElectroJr) [reply Content.Server/Construction/AnchorableSystem.cs:93]: https://github.com/space-wizards/RobustToolbox/pull/3634 probably fixes this.

**Comment by eoineoineoin** [comment]
https://github.com/space-wizards/RobustToolbox/pull/3634 does not reproduce the issue anymore, so closing this PR.

**Pull request closed without merging**